### DISCUSSION TITLE
Python Module is not installing !!

### DISCUSSION THREAD

**Discussion by tanujdhiman** [apache/airflow#17658] [Q&A]
**Apache Airflow version**: 1.10.14

**OS**: WSL Ubuntu 

**Apache Airflow Provider versions**:

**Deployment**: Using cmd `docker compose up`

**What happened**:
Actually I am installing `pydub` package in WSL Ubuntu with apache airflow. But When I start server with `localhost:8080` then I am getting error that module is not installed before I start server I installed `sudo pip install pydub` and also module is perfectly installed with no errors but after running found this :

![1](https://user-images.githubusercontent.com/56601466/129728096-93987ff3-360d-4aa6-9097-89e8ce8af2e2.PNG)

**What you expected to happen**:
I want apache airflow server running without any erros.

**How to reproduce it**:

**Anything else we need to know**:

**Are you willing to submit a PR?**

**Comment by jedcunningham** [comment]
@tanujdhiman, you mention using `docker-compose` for your deployment method, but you are trying to install the package on the host machine directly. That won't work. You need to install the package in the docker image. Here are docs on how to achieve it: https://airflow.apache.org/docs/docker-stack/build.html#example-of-adding-pypi-package

(I've deep linked you to the exact example, but reading through the whole page might be worthwhile)

**Reply by tanujdhiman** [reply]
Can you give any example for this?

**Reply by tanujdhiman** [reply]
I'm talking about what command should I run which properly install this.

**Comment by LeonZhao28** [comment]
It is useless for you to install pydub before start the server.
The airflow is running in the docker and the `pydub` is installed on the host machine.
You can build your own docker image based on the official airflow docker image and change the docker image in the file `docker-compose.yaml`
Or mount your local python `site-packages` into the docker.

**Reply by LeonZhao28** [reply]
or you can use PythonVirtualenvOperator  instead of PythonOperator
the doc link is :https://airflow.apache.org/docs/apache-airflow/stable/howto/operator/python.html

**Reply by potiuk** [reply]
[Building your own image](https://airflow.apache.org/docs/docker-stack/build.html) is best when you productionize it. For "quick" bootstrapping and testing you can also use _PIP_ADDITIONAL_REQUIREMENTS variable - https://airflow.apache.org/docs/apache-airflow/stable/start/docker.html#environment-variables-supported-by-docker-compose . It is not good for production (again building image is the only really feasible approach) but if you want to just play and iterate with different dependencies, it is a bit faster to iterate than building image.
